Exercises/NoModules/Chapter 16/Soln16_06/DB_RAII.h

```diff
@@ -0,0 +1,68 @@
+// RAII classes for handles returned by DB.h C interface functions
+
+/*
+  In this solution we have the RAII classes accept the resource handle in their constructor.
+  Alternatively, you could also acquire the resources inside the constructors of the RAII class.
+  For instance: you could call db_connect() from within the DBConnectionRAII() contructor.
+*/
+
+#ifndef DB_RAII_H
+#define DB_RAII_H
+
+#include "DB.h"
+
+/**
+ * RAII object that ensures that a given database connection is closed
+ * once the RAII object goes out of scope.
+ */ 
+class DBConnectionRAII
+{
+public:
+  DBConnectionRAII(DB_CONNECTION* connection) noexcept
```

Comment:
explicit

Reply:
For once I'm ok with this not being explicit I think...